packages/utils/upgrade/README.md

```diff
@@ -1 +1,179 @@
-<h1 align="center">upgrade</h1>
+# Strapi Upgrade Tool
+
+## Description
+
+The Strapi Upgrade Tool is the CLI for facilitating upgrades between Strapi versions.
+
+When run, the command will "step" from your current version of Strapi to the next version that includes code tranforms for breaking changes.
+
+From now on, all breaking changes, and ideally also deprecations, must have an accompanying code transform written to be accepted.
+
+### Types of Transforms
+
+The upgrade tool provides two types of transforms:
+
+- `json`: for updating a project's json files, primarily intended for the `package.json`
+- `code`: codemods; for updating a project's [js, jsx, ts, tsx] files
+
+### What's a codemod?
+
+Codemods are a scripted way to refactor code. Here we are providing and running these scripts for users for any changes necessary in user code between Strapi versions.
+
+For example, if we need to rename a package used by Strapi projects, instead of instructing users to change the import, we provide a script that searches through the user's project and does the replacement for them.
+
+### Data Migrations
+
+Data migrations are not handled by the upgrade tool.
+
+For Strapi v4, no data migrations will be allowed and no support is planned (except in extenuating circumstances eg, a critical security issue somehow relating to the database shape)
+
+For Strapi v5, automated data migrations can be added in the `packages/core/database` package of the `v5/main` branch of this repo.
+
+## Usage
+
+This package is not yet released, so currently it can be run on a project in the monorepo /examples directory with the following command:
+
+`../../packages/utils/upgrade/bin/upgrade`
+
+Run the command with the `--help` option to see all the available options.
+
+[Coming Soon] The Strapi Upgrade tool will be available using `npx strapi-upgrade`
```

Comment:
Shouldn't it be `@strapi/upgrade` ?

Reply:
Also, we do plan to add shortcuts to the strapi CLI (e.g. `yarn strapi upgrade`). I don't know if that's worth to mention here